File: engine/model/src/main/scala/tools/packet/family/FamilyLogonNotice.scala
package tools.packet.family

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class FamilyLogonNotice(private var _characterName: String, private var _loggedIn: Boolean) extends PacketInput {
  def characterName: String = _characterName

  def loggedIn: Boolean = _loggedIn

  override def opcode(): SendOpcode = SendOpcode.FAMILY_NOTIFY_LOGIN_OR_LOGOUT
}

File: engine/engine-model/src/main/java/tools/packet/family/FamilyLogonNotice.java
package tools.packet.family;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record FamilyLogonNotice(String characterName, Boolean loggedIn) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.FAMILY_NOTIFY_LOGIN_OR_LOGOUT;
   }
}